plugins/modules/aci_cloud_ctx_profile.py

```diff
@@ -0,0 +1,234 @@
+    name: ctx_profile_1
+    state: query
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        {
+            'descr': dict(type='str',),
+            'name': dict(type='str', aliases=['cloud_context_profile']),
+            'nameAlias': dict(type='str',),
+            'type': dict(type='str', choices=['regular', 'shadow'], ),
+            'tenant': dict(type='str',),
+            'state': dict(type='str', default='present', choices=['absent', 'present', 'query']),
+            'primary_cidr': dict(type='str',),
+            # FIXME: didn't find the flow_log in UI
+            # 'flow_log': dict(type='str'),
+            'vrf': dict(type='str'),
+            'region': dict(type='str'),
+            'vpn_gateway': dict(type='bool', default=False)
+        }
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['name', 'tenant']],
+            ['state', 'present', ['name', 'tenant', 'vrf', 'region', 'primary_cidr']],
+        ],
+    )
+
+    descr = module.params['descr']
+    name = module.params['name']
+    nameAlias = module.params['nameAlias']
+    type = module.params['type']
+    tenant = module.params['tenant']
+    state = module.params['state']
+    primary_cidr = module.params['primary_cidr']
+    child_configs = []
+
+    vrf = module.params['vrf']
+    region = module.params['region']
+    vpn_gateway = module.params['vpn_gateway']
+    if vrf:
+        child_configs.append({'cloudRsToCtx': {'attributes': {'tnFvCtxName': vrf}}})
+    if region:
+        child_configs.append(
+            {
+                'cloudRsCtxProfileToRegion': {
+                    'attributes': {
+                        'tDn': "uni/clouddomp/provp-aws/region-{0}".format(region)
+                    }
+                }
+            }
+        )
+    if vpn_gateway:
+        child_configs.append(
+            {
+                "cloudRouterP": {
+                    "attributes": {
+                        "name": "default"
+                    },
+                    "children": [{
+                        "cloudIntNetworkP": {
+                            "attributes": {
+                                "name": "default"
+                            }
+                        }
+                    }]
+                }
+            }
+        )
+    child_configs.append(
+        {
+            'cloudCidr': {
+                'attributes': {
+                    "addr": primary_cidr,
+                    "primary": "yes"
+                }
+            }
+        }
+    )
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class={
+            'aci_class': 'fvTenant',
```

Comment:
in almost all the modules, this is represented with a dict like:
root_class=dict(
            aci_class=domain_class,
            aci_rn=domain_rn,
            module_object=domain_mo,
            target_filter={'name': domain},
        ),

I think we should stick to common representation.